### PR TITLE
[TC-5357] Support destructive alert style

### PR DESCRIPTION
**Pull request by hansemannn**
JIRA reference: https://jira.appcelerator.org/browse/TC-5357

### CODE REVIEW

File: iphone/Classes/TiUIAlertDialogProxy.m [hunk 1]
```diff
@@ -134,7 +136,7 @@ -(void)show:(id)args
                 NSString* btnName = [TiUtils stringValue:btn];
                 if (!IS_NULL_OR_NIL(btnName)) {
                     UIAlertAction* theAction = [UIAlertAction actionWithTitle:btnName
-                                                                        style:((curIndex == cancelIndex) ? UIAlertActionStyleCancel : UIAlertActionStyleDefault)
+                                                                        style:((curIndex == cancelIndex) ? UIAlertActionStyleCancel : (curIndex == destructiveIndex) ? UIAlertActionStyleDestructive :UIAlertActionStyleDefault)
```
Comment (pec1985) [inline iphone/Classes/TiUIAlertDialogProxy.m]: This line is a bit confusing and not easy to follow due to the multiple ternary operators. If would be better to spread the out for clarity purposes, something like this:
```
UIAlertActionStyle style;
if (curIndex == cancelIndex) {
	style = UIAlertActionStyleCancel;
} else if (curIndex == destructiveIndex) {
	style = UIAlertActionStyleDestructive;
} else {
	style  = UIAlertActionStyleDefault;
}
```

File: apidoc/Titanium/UI/AlertDialog.yml [hunk 2]
```diff
@@ -168,6 +168,14 @@ properties:
...
     type: Number
     default: undefined (Android), -1 (iOS, Mobile Web)
+
+  - name: destructive
+    summary: Index to define the destructive button.
+    platforms: [iphone, ipad]
+    description: |
+        Note that this property available on iOS SDK 8 or above. 
+    type: Number
+    default: -1
```
Comment (pec1985) [inline apidoc/Titanium/UI/AlertDialog.yml]: Would be nice to specify that setting it to `-1` disables this option, or does nothing. Having a default value of `-1` without explaining what it does makes it unclear

**Comment by pec1985** [comment]
PR reviewed, left a couple of comments.

File: iphone/Classes/TiUIAlertDialogProxy.m (hunk 1 above)
Comment (pec1985) [inline iphone/Classes/TiUIAlertDialogProxy.m]: Also, if this is iOS 8 an above only, make sure you check for that with:
```
if ([TiUtils isIOS8OrGreater]) { ..... }
```

**Comment by hansemannn** [comment]
Thank you! Comments are reasonable, will have a look at it!

**Comment by hansemannn** [comment]
Updated docs and code readability as well as a bounding-check like it is done with the cancel-button. I think the iOS 8 check is not necessary in this case, because the code is already in the conditional iOS 8 block to support the UIAlertController.

**Comment by pec1985** [comment]
One more thing... the destructive index is missing from the click event. For consistency, add it here as well `fireClickEventWithAction`:
```
NSMutableDictionary *event = [NSMutableDictionary dictionaryWithObjectsAndKeys:
                               NUMUINTEGER(indexOfAction),@"index",
                               [NSNumber numberWithInt:cancelIndex],@"cancel",
                               nil];
```

**Comment by hansemannn** [comment]
Added the destructive index and implemented a quick sample example code to test:

``` javascript
var win = Ti.UI.createWindow({
	backgroundColor: "#fff"
});

var button = Ti.UI.createButton({
	title: "Show destructive dialog"
});

var dialog = Ti.UI.createAlertDialog({
	title: "Proceed",
	message: "Do you really want to proceed?",
	destructive: 0,
	cancel: 1,
	buttonNames: ["Destructive", "Cancel", "Proceed"]
});

button.addEventListener("click", function() {
	dialog.show();
});

dialog.addEventListener("click", function(e) {
	Ti.API.info(e);
});

win.add(button);

win.open();
```

**Comment by pec1985** [comment]
i have opened a new PR with your commits for Titanium master branch #6842

**Comment by pec1985** [comment]
There is one minor thing missing in the documentation. The `since` tag (for Ti SDK) and the `osver` for the iOS version. You can simple cherry pick from master since I already merged that
```
git cherry-pick cfcbc96
```

**Comment by hansemannn** [comment]
Closing this outdated PR in favor of #6842

**Pull request closed without merging**